src/esp/scene/SceneNode.h

```diff
@@ -335,6 +334,29 @@ void preOrderFeatureTraversalWithCallback(SceneNode& node, Callable&& cb) {
       const_cast<const SceneNode&>(node), constFeatureCb);
 }
 
+/**
+ * @brief Set the semantic ID of a scene graph subtree.
+ *
+ * @param node Root node of the subtree.
+ * @param semanticId Semantic ID to apply to the subtree.
+ */
+static inline void setSemanticIdForSubtree(esp::scene::SceneNode* node,
```

Comment:
No `static` for free functions here either, thanks ;)

And I'd put it into the `*.cpp` as well, it doesn't need to be in the header and you save a bit on compile times that way.